crates/sui-framework/packages/dwallet-system/sources/dwallet_2pc_mpc_ecdsa_k1.move

```diff
@@ -0,0 +1,302 @@
+
+    #[allow(unused_field)]
+    struct PresignSession has key {
+        id: UID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        hash: u8,
+        commitments_and_proof_to_centralized_party_nonce_shares: vector<u8>,
+        messages: vector<vector<u8>>,
+        sender: address,
+    }
+
+    #[allow(unused_field)]
+    struct PresignSessionOutput has key {
+        id: UID,
+        session_id: ID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        output: vector<u8>,
+    }
+
+    #[allow(unused_field)]
+    struct Presign has key {
+        id: UID,
+        session_id: ID,
+        dwallet_id: ID,
+        dwallet_cap_id: ID,
+        presigns: vector<u8>,
+    }
+
+    #[allow(unused_field)]
+    struct SignData has store {
+        presign_session_id: ID,
+        hash: u8,
+        public_nonce_encrypted_partial_signature_and_proofs: vector<u8>,
+        presigns: vector<u8>,
+    }
+
+    public fun create_dkg_session(commitment_to_centralized_party_secret_key_share: vector<u8>, ctx: &mut TxContext): DWalletCap {
+        let cap = create_dwallet_cap(ctx);
+        let sender = tx_context::sender(ctx);
+        let session = DKGSession {
+            id: object::new(ctx),
+            dwallet_cap_id: object::id(&cap),
+            commitment_to_centralized_party_secret_key_share,
+            sender,
+        };
+        event::emit(NewDKGSessionEvent {
+            session_id: object::id(&session),
+            dwallet_cap_id: object::id(&cap),
+            commitment_to_centralized_party_secret_key_share,
+            sender,
+        });
+        transfer::freeze_object(session);
+        cap
+    }
+
+    #[allow(unused_function)]
+    fun create_dkg_output(session: &DKGSession, commitment_to_centralized_party_secret_key_share: vector<u8>, secret_key_share_encryption_and_proof: vector<u8>, ctx: &mut TxContext) {
+        assert!(tx_context::sender(ctx) == @0x0, ENotSystemAddress);
+        let output = DKGSessionOutput {
+            id: object::new(ctx),
+            session_id: object::id(session),
+            dwallet_cap_id: session.dwallet_cap_id,
+            commitment_to_centralized_party_secret_key_share,
+            secret_key_share_encryption_and_proof
+        };
+        transfer::transfer(output, session.sender);
+    }
+
+    public fun create_dwallet(output: DKGSessionOutput, centralized_party_public_key_share_decommitment_and_proof: vector<u8>, ctx: &mut TxContext) {
+        let DKGSessionOutput {
+            id,
+            session_id,
+            dwallet_cap_id,
+            commitment_to_centralized_party_secret_key_share,
+            secret_key_share_encryption_and_proof,
+        } = output;
+        object::delete(id);
+
+        let (output, public_key) = dkg_verify_decommitment_and_proof_of_centralized_party_public_key_share(commitment_to_centralized_party_secret_key_share, secret_key_share_encryption_and_proof, centralized_party_public_key_share_decommitment_and_proof);
```

Comment:
what happens if verification of zk proofs fails? does the native function fail with an exception that aborts the execution of this function?